mllib/src/main/scala/org/apache/spark/mllib/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -107,8 +107,11 @@ private[tree] object DecisionTreeMetadata extends Logging {
       numTrees: Int,
       featureSubsetStrategy: String): DecisionTreeMetadata = {
 
-    val numFeatures = input.take(1)(0).features.size
+    require(!input.isEmpty, s"DecisionTree requires size of input RDD > 0, " +
```

Comment:
This will trigger `take(1)` twice. Should be

~~~
val numFeatures = input.map(_.features.size).take(1).headOption.getOrElse {
  throw new IllegalArgumentException("...")
}
~~~